src/server/authMiddleware.ts

```diff
@@ -25,6 +25,10 @@ type MiddlewareOptions = {
 	// - process.env.SIGN_IN_ROUTE or /sign-in if not provided
 	// - process.env.SIGN_UP_ROUTE or /sign-up if not provided
 	publicRoutes?: string[];
+
+	// An array of private routes that require authentication
+	// If privateRoutes is defined, routes not listed in this array will default to public routes
+	privateRoutes?: string[];
```

Comment:
I like the idea but I think we need to consider the hierarchy here and if it make sense to pass both public and private routes (I don't think it make sense to pass both of them) 

We also need to reflect that with TS (allow only one of them and not both)

Reply:
I made this change.